lumicks/pylake/detail/confocal.py

```diff
@@ -148,6 +149,36 @@ def from_json(cls, json_string):
         return cls(axes, json_dict["scan volume"]["center point (um)"], json_dict["scan count"])
 
 
+def _deprecate_basescan_plot_args(plot):
+    """Decorator to deprecate second positional argument 'axes' of the method `BaseScan.plot()`"""
+    import functools
+    import warnings
+    import matplotlib
+
+    @functools.wraps(plot)
+    def wrapper(self, *args, **kwargs):
+        # The plot function might be called with up to 2 positional arguments. We can gracefully
+        # convert the second positional argument (axes) into a keyword argument.
+        # old arguments: channel, axes
+        if len(args) == 2 and (isinstance(args[1], matplotlib.axes.Axes) or args[1] is None):
```

Comment:
I think this should handle the case `len(args) > 2` (see below). The annoying part is that the signature for `Scan` and `Kymo` are different. The positional args for `Scan` and `Kymo` are `channel, axes, [frame, handle], adjustments, other kwargs` between `[]` are `Scan` exclusive.